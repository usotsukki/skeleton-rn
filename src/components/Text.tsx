import { Text as UIKitText } from 'react-native-ui-lib'
import { TextPropTypes } from 'react-native-ui-lib/src/components/text'

const Text = (props: TextPropTypes) => {
	return <UIKitText {...props}>{props.children}</UIKitText>
}

export default Text
