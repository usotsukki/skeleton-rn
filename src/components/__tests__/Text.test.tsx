import { render } from '@testing-library/react-native'
import React from 'react'
import { TextDriver } from 'react-native-ui-lib/testkit'
import Text from '../Text'

it(`renders correctly`, () => {
	const renderTree = render(<Text testID="text">Snapshot test?</Text>)
	const textNode = TextDriver({ renderTree, testID: 'text' })
	expect(textNode).toMatchSnapshot()
})
