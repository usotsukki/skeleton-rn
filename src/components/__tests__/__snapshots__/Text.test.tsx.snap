// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Text
  style={
    [
      {
        "backgroundColor": "transparent",
        "color": "#20303C",
        "writingDirection": "ltr",
      },
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
    ]
  }
  testID="text"
>
  Snapshot test?
</Text>
`;
